From: AoHui <aohui.wan@gmail.com>
Date: Fri, 10 Feb 2023 16:20:20 +0000
Subject: Fix "gn gen" failure from unused "_module_target_name"

Bug: None
Signed-off-by: AoHui <aohui.wan@gmail.com>
Change-Id: I9b12f821082204c4ca246b8dd5c991b67d54f35b
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/4238336
Reviewed-by: Egor Pasko <pasko@chromium.org>
Commit-Queue: Andrew Grieve <agrieve@chromium.org>
Cr-Commit-Position: refs/heads/main@{#1103881}
---
 AUTHORS                        | 1 +
 build/config/android/rules.gni | 2 +-
 2 files changed, 2 insertions(+), 1 deletion(-)

diff --git a/AUTHORS b/AUTHORS
--- a/AUTHORS
+++ b/AUTHORS
@@ -126,6 +126,7 @@ Anton Obzhirov <a.obzhirov@samsung.com>
 Antonin Hildebrand <antonin.hildebrand@gmail.com>
 Antonio Gomes <a1.gomes@sisa.samsung.com>
 Anuj Kumar Sharma <anujk.sharma@samsung.com>
+Ao Hui <aohui.wan@gmail.com>
 Ao Sun <ntusunao@gmail.com>
 Ao Wang <wangao.james@bytedance.com>
 Aquibuzzaman Md. Sayem <md.sayem@samsung.com>
diff --git a/build/config/android/rules.gni b/build/config/android/rules.gni
--- a/build/config/android/rules.gni
+++ b/build/config/android/rules.gni
@@ -5074,11 +5074,11 @@ if (enable_java_templates && is_android) {
     _all_module_unused_resources_deps = []
     foreach(_module, _modules) {
       _module_target = _module.module_target
-      _module_target_name = get_label_info(_module_target, "name")
       _module_build_config = _module.build_config
       _module_build_config_target = _module.build_config_target
 
       if (!_proguard_enabled) {
+        _module_target_name = get_label_info(_module_target, "name")
         _dex_target = "${_module_target_name}__final_dex"
         _rebased_module_build_config =
             rebase_path(_module_build_config, root_build_dir)
-- 
2.39.2

